packages/metal-dom/src/dom.js

```diff
@@ -306,28 +306,32 @@ class dom {
 	 * @protected
 	 */
 	static handleDelegateEvent_(event) {
-		dom.normalizeDelegateEvent_(event);
 		var currElement = core.isDef(event[NEXT_TARGET]) ?
-			event[NEXT_TARGET] :
-			event.target;
+				event[NEXT_TARGET] :
+				event.target;
 		var ret = true;
-		var container = event.currentTarget;
-		var limit = event.currentTarget.parentNode;
-		var defFns = [];
 
-		while (currElement && currElement !== limit && !event.stopped) {
-			event.delegateTarget = currElement;
-			ret &= dom.triggerMatchedListeners_(container, currElement, event, defFns);
-			currElement = currElement.parentNode;
-		}
+		if (dom.isAbleToInteractWith_(currElement, event.type)) {
```

Comment:
If you have time then sending everything would be great :) But just make sure you add the test and I'll merge, even if the refactoring comes separately later.